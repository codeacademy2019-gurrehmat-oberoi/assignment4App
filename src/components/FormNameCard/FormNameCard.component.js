import React, { Component } from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from './FormNameCard.component.style';

class IdeaCard extends Component {
  render() {
    const {
      title,
      date,
    } = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{title}</Text>
        <View style={styles.hr} />
        <Text style={styles.date}>{`Created on: ${date}`}</Text>
      </View>
    );
  }
}

IdeaCard.propTypes = {
  title: PropTypes.string,
  date: PropTypes.instanceOf(Date),
};

IdeaCard.defaultProps = {
  title: 'Default Title',
  date: new Date(2019, 1, 1),
};

export default IdeaCard;
