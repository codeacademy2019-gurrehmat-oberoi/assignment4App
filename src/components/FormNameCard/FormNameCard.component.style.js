import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10,
    borderWidth: 2,
    backgroundColor: '#FFFFFF',
    // alignItems: 'stretch',
  },
  title: {
    fontSize: 30,
    alignSelf: 'center',
    textAlign: 'center',
  },
  hr: {
    margin: 5,
    backgroundColor: 'black',
    height: 2,
  },
  date: {
    fontSize: 20,
    alignSelf: 'flex-end',
  },
});

export default styles;
