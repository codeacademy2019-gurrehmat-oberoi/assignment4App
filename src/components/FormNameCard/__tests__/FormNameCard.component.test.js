import React from 'react';
import renderer from 'react-test-renderer';
import FormNameCard from '../FormNameCard.component';

describe('Form Name Card', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<FormNameCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
