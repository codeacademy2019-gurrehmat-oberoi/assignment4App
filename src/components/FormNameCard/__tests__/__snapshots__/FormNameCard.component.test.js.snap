// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Name Card should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderWidth": 2,
      "flex": 1,
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 30,
        "textAlign": "center",
      }
    }
  >
    Default Title
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "height": 2,
        "margin": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "fontSize": 20,
      }
    }
  >
    Created on: Fri Feb 01 2019 00:00:00 GMT+0530 (India Standard Time)
  </Text>
</View>
`;
